overlord/hookstate/context.go

```diff
@@ -127,3 +133,57 @@ func (c *Context) Get(key string, value interface{}) error {
 
 	return nil
 }
+
+// State returns the state contained within the context
+func (c *Context) State() *state.State {
+	return c.task.State()
+}
+
+// Cached returns the cached value associated with the provided key.
+// It returns nil if there is no entry for key.
+func (c *Context) Cached(key interface{}) interface{} {
+	cachedCache := c.task.State().Cached(contextCache{})
```

Comment:
sorry, I didn't notice the first time around but it means one cache shared by all contexts, that's not what we want I think, and indeed as Gustavo pointed out nothing will remove this stuff

we need the full Cache/Cached-like logic copied from state.State